### PR TITLE
fix(laravel): allow boolean cast

### LINKED ISSUE #6807: [Laravel] Boolean Cast is not working with GraphQL
**API Platform version(s) affected**: 4.*

**Description**  
Adding a boolean cast doesn't work correctly with GraphQL as it requires String! type

**How to reproduce**  
```
    protected $casts = [
        'id' => 'integer',
        'is_active' => 'boolean',
    ];
```

### PR DESCRIPTION
**Pull request by amermchaudhary**
| Q             | A
| ------------- | ---
| Branch?       | 4.0
| Tickets       | Closes #6807
| License       | MIT

### CODE REVIEW

**Review by dunglas — APPROVED** [review]

**Comment by amermchaudhary** [comment]
@dunglas I created a test for this but the test is failing because of another bug. I will add that bug and try to fix it as well. Basically I tried to add a create mutation for the book but the mutation is not working because its expecting an author of type string.

**Comment by amermchaudhary** [comment]
@dunglas Nevermind, that was my mistake. Tests have been added. Additionally I also see a problem with array casting. Should I fix it in the same PR or do you want me to create a separate bug and corresponding PR?

**Comment by dunglas** [comment]
Adding the fix and the test in this PR, or creating a new one are both OK.

Don't bother opening a new issue, the describing the fix in the PR description is enough.

Thank you for these fixes!

**Review by dunglas — APPROVED** [review]

File: src/Laravel/Tests/GraphQlTest.php [hunk 1]
```diff
@@ -66,4 +66,38 @@ public function testGetBooksWithPaginationAndOrder(): void
...
         $this->assertArrayNotHasKey('errors', $data);
     }
+
+    public function testCreateBook(): void
+    {
+        /** @var \Workbench\App\Models\Author $author */
+        $author = AuthorFactory::new()->create();
+        $response = $this->postJson('/api/graphql', [
+            'query' => '
+                mutation createBook($book: createBookInput!){
```
Comment (dunglas) [inline src/Laravel/Tests/GraphQlTest.php:76]: Nit: you could use nowdoc strings for readability.

**Comment by soyuka** [comment]
Thanks!

**Pull request merged**